cpp/src/arrow/array/data.h

```diff
@@ -434,6 +434,12 @@ struct ARROW_EXPORT ArraySpan {
     return GetValues<T>(i, this->offset);
   }
 
+  // Access a buffer's data as a span
+  template <typename T>
+  util::span<const T> GetSpan(int i) const {
+    return util::span(buffers[i].data_as<T>(), static_cast<size_t>(buffers[i].size));
```

Comment:
This needs to account for offset as well as divide the buffer's remaining size by `sizeof(T)`

Reply:
Maybe like so:
```suggestion
    return util::span(buffers[i].data_as<T>(), static_cast<size_t>(buffers[i].size) / sizeof(T)).subspan(offset);
```

Reply:
thanks!